lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb

```diff
@@ -329,32 +330,36 @@ def run_sync(command, stdin = nil)
       stdout = ''
       stderr = ''
       exit_status = nil
-      started = false
 
-      channel = session.open_channel do |ch|
-        ch.on_data do |c, data|
-          stdout.concat(data)
-        end
-        ch.on_extended_data { |_, _, data| stderr.concat(data) }
-        ch.on_request('exit-status') { |_, data| exit_status = data.read_long }
-        # Send data to stdin if we have some
-        ch.send_data(stdin) unless stdin.nil?
-        # on signal: sending the signal value (such as 'TERM')
-        ch.on_request('exit-signal') do |_, data|
-          exit_status = data.read_string
-          ch.close
-          ch.wait
-        end
-        ch.exec command do |_, success|
-          raise 'could not execute command' unless success
-
-          started = true
-        end
+      options = []
+      options << "-o User=#{@ssh_user}"
+      options << "-o Port=#{ssh_options[:port]}" if ssh_options[:port]
+      options << "-o IdentityFile=#{ssh_options[:keys][0]}" if ssh_options[:keys]
+      options << "-o IdentitiesOnly=yes"
+      options << "-o StrictHostKeyChecking=accept-new"
+      options << "-o PreferredAuthentications=#{ssh_options[:auth_methods].join(',')}"
+      options << "-o UserKnownHostsFile=#{ssh_options[:user_known_hosts_file]}" if ssh_options[:user_known_hosts_file]
+      options << "-o NumberOfPasswordPrompts=#{ssh_options[:number_of_password_prompts]}"
+      options << "-o LogLevel=#{ssh_options[:verbose]}"
+
+      master, slave = PTY.open
+      read, write = IO.pipe
+      pid = spawn({"SSHPASS" => ssh_options[:password]}, "/usr/bin/sshpass", "-e", "/usr/bin/ssh", "#{@host}", *options, command, :in => read, :out => slave)
```

Comment:
Why do you need string interpolation on `@host`?
```suggestion
      pid = spawn({"SSHPASS" => ssh_options[:password]}, "/usr/bin/sshpass", "-e", "/usr/bin/ssh", @host, *options, command, :in => read, :out => slave)
```